### ISSUE TITLE
Subscribe to run events on RunsHistory

### ISSUE THREAD

**Issue by taylordowns2000** [OpenFn/Lightning#490]
backing out to a purely non-technical user perspective:
- i want to know the status of runs… particularly those that are running _right now_! they’re the most important as i’m building stuff.
- when i’m looking at that history page, i want it to always reflect the current state of those runs (and subsequently the attempts and work-orders, whose “state” is made up of aggregate run states) without having to either refresh my page or click away and click back.

**Additional context**

Run events are related to runs, and we should subscribe to `new` and `update` events. But when we get info on a run, we need to then update the attempt and work order, which include calculated aggregates of runs.

Beyond that, if we get data about a new or updated run, we might not yet have data on the work order or the attempt. We need to figure out whether to add a new WO or attempt to the page or ignore it, because it's related to a work order or attempt that's hidden by the current filters or pagination.

In other words, this should work just like it does on openfn.org in the runs list.
